saleor/graphql/payment/mutations.py

```diff
@@ -198,6 +237,91 @@ def perform_mutation(cls, _root, info, checkout_id=None, token=None, **data):
         return CheckoutPaymentCreate(payment=payment, checkout=checkout)
 
 
+class CheckoutPaymentComplete(BaseMutation, I18nMixin):
+    checkout = graphene.Field(Checkout, description="Related checkout object.")
+    payment = graphene.Field(Payment, description="Related payment object.")
+    confirmation_needed = graphene.Boolean(
+        required=True,
+        default_value=False,
+        description=(
+            "Set to true if payment needs to be confirmed"
+            " before checkout is complete."
+        ),
+    )
+    confirmation_data = graphene.JSONString(
+        required=False,
+        description=(
+            "Confirmation data used to process additional authorization steps."
+        ),
+    )
+
+    class Arguments:
+        token = UUID(required=True, description="Checkout token.")
+        payment_id = graphene.ID(required=True, description="Payment ID.")
+        payment_data = graphene.JSONString(
+            required=False,
+            description=(
+                "Client-side generated data required to finalize the payment."
+            ),
+        )
+        redirect_url = graphene.String(
+            required=False,
+            description=(
+                "URL of a view where users should be redirected to "
+                "see the order details. URL in RFC 1808 format."
+            ),
+        )
+
+    class Meta:
+        description = (
+            f"{ADDED_IN_31} Completes an individual payment as part of the checkout. "
+            "This mutation does not create the order and any webhooks related "
+            "to this payment will neither create the order."
+            "In case an additional confirmation step such as 3D secure is required "
+            "confirmationNeeded flag will be set to True and no authorization will be "
+            "held until the payment is confirmed with second call of this mutation."
+        )
+        error_type_class = CheckoutError
+        error_type_field = "checkout_errors"
+
+    @classmethod
+    def perform_mutation(cls, _root, info, token, payment_id, **data):
+        tracking_code = analytics.get_client_id(info.context)
+        with transaction_with_commit_on_errors():
+            checkout = get_checkout_by_token(token)
```

Comment:
Shouldn't you lock the checkout in the same way as you did for checkout_complete mutation?